src/ssl/support.cc

```diff
@@ -390,9 +390,35 @@ ssl_verify_cb(int ok, X509_STORE_CTX * ctx)
 }
 
 void
-Ssl::SetupVerifyCallback(Security::ContextPointer &ctx)
+Ssl::ConfigurePeerVerification(Security::ContextPointer &ctx, int flags)
 {
-    SSL_CTX_set_verify(ctx.get(), SSL_VERIFY_PEER | SSL_VERIFY_FAIL_IF_NO_PEER_CERT, ssl_verify_cb);
+    int mode;
+
+    if(flags & SSL_FLAG_DONT_VERIFY_PEER) {
+        debugs(83, 2, "SECURITY WARNING: Peer certificates are not verified for validity!");
+        mode = SSL_VERIFY_NONE;
+    }
+    else if(flags & SSL_FLAG_DELAYED_AUTH) {
+        debugs(83, 9, "Not requesting client certificates until acl processing requires one");
```

Comment:
Please use a level 3-6 for messages that may be of use to admin, but not important enough to display by default. As Alex already mentioned 9 is reserved for huge data dumps.

You can also use DBG_PARSE_NOTE() to inform admin doing a config test what is going to happen during production.
```suggestion
        debugs(83, DBG_PARSE_NOTE(3), "not requesting client certificates until ACL processing requires one");
```

Reply:
I simply copied what was already there.